### ISSUE TITLE
[UI] TFMA visualization loads slowly

### ISSUE THREAD

**Issue by Bobgy** [kubeflow/pipelines#3114]
Part of https://github.com/kubeflow/pipelines/issues/2283

### What steps did you take:
1. visit KFP UI -> run details page -> node details
2. use custom visualization to run tfma visualization

### What happened:
visualization loads slowly with some too many redirect errors like the following:
<img width="1037" alt="Screen Shot 2020-02-19 at 3 04 34 PM" src="https://user-images.githubusercontent.com/4957653/74809921-3d928b00-5329-11ea-897a-592f4d1d5bc4.png">

### What did you expect to happen:
There shouldn't be too many redirect errors, but scripts should be loaded from cdn directly.

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?

hosted

### Anything else you would like to add:
[Miscellaneous information that will assist in solving the issue.]

/kind bug
/area frontend

**Comment by Bobgy** [comment]
I'm found root cause for the infinite redirection is:
1. requirejs loads modules from local path first, so it tries to load tfma_widget_js and @jupyter/html-manager from https://281b9edc3093660f-dot-asia-east1.pipelines.googleusercontent.com/ (hosted kfp base path)
2. however, the script runs in a sandboxed iframe, so it cannot use credentials to bypass base path proxy authentication
3. therefore, base path redirects to google login page
4. however, the login page request somehow can actually send credentials, so it just successfully verifies credentials and redirects back to kfp host base path
5. no surprise, kfp host base path still cannot authenticate, so this creates a loop of redirection.

However, ... all of these shouldn't happen in the first place, because the js files should be loaded from external cdn directly.

**Comment by Bobgy** [comment]
A second problem is that, this `@jupyter/html-manager` should already load if user code uses the following
```
from ipywidgets.embed import embed_minimal_html

analysis_path = 'gs://<TFMA_OUTPUT_DIRECTORY>'
result = tfma.load_eval_result(output_path=analysis_path)
slicing_metrics_view = tfma.view.render_slicing_metrics(result)
embed_minimal_html('tfma_export.html', views=[slicing_metrics_view], title='Slicing Metrics')
```

**Comment by Bobgy** [comment]
The unnecessary html-manager comes from https://github.com/kubeflow/pipelines/blob/7ee3244f5b07614d76213c1cc59fcef75f46556c/backend/src/apiserver/visualization/templates/full.tpl#L21 which I believe doesn't load in current use cases.

@ajchili Can you confirm if this is true?

**Comment by Bobgy** [comment]
I'm going to verify if removing the addWidgetsRenderer can still allow current supported visualizations for TFDV and TFMA. We can add better support for this later.

**Comment by Bobgy** [comment]
Verified tfma visualization still works without addWidgetsRenderer, and saves about 2 seconds from loading.

**Comment by Bobgy** [comment]
Filed an upstream feature request for @jupyter/html-manager to fix this issue: https://github.com/jupyter-widgets/ipywidgets/issues/2786

**Comment by Bobgy** [comment]
My upstream PR is merged, I will wait for ipywidgets 8.0 release to switch to use my new feature

**Comment by Bobgy** [comment]
#3712 is pretty much all we can do now, integrating with new (not yet released) ipywidgets 8.0 can be post 1.0

**Comment by rimolive** [comment]
Closing this issue. No activity for more than a year.
